### ISSUE TITLE
Change ColorPicker template colors

### ISSUE THREAD

**Issue by ClayYahuira** [AvaloniaUI/Avalonia#14145]
## Describe the bug
I'm trying to change the color of the PathIcon inside the flyout on each tab but is not working properly. The Foreground color that I set is working when I open the color picker in the second time, in the first time is showing the default color.

## To Reproduce

1. Add a ColorPicker in the project.
2. Tap over the ColorPicker.
3. You will see the default Foreground color in the PathIcon (TabItem.Header > Border > PathIcon).
4. Close the ColorPicker.
5. Open the ColorPicker again.
6. You will se the setted Foreground color in the PathIcon.

## Expected behavior
The PathIcon should show the setted Foreground color since the first time.

## Environment 

- OS: Android, iOS, Mac
- Avalonia-Version: 11.0.4

## Additional context
This is the way I'm setting the PathIcon Foreground:

        ```
<Style Selector="ColorPicker /template/ PathIcon">
                <Setter Property="Foreground" Value="#7FFEC2" />
        </Style>
```

**Comment by ClayYahuira** [comment]
```
 <Style Selector="ColorPicker /template/ PathIcon">
                <Setter Property="Foreground" Value="#7FFEC2" />
        </Style>
```
I'm setting the Foreground with this code.

**Comment by ClayYahuira** [comment]
> Can you please also try on ColorView? I think it might relate to popup.

I tried aplying the style to the ColorView but it just work for the ColorView, this is not aplying to the ColorView in the popup.

**Comment by robloo** [comment]
>>Can you please also try on ColorView? I think it might relate to popup.
>
>I tried aplying the style to the ColorView but it just work for the ColorView, this is not aplying to the ColorView in the popup.
I was checking the .xaml file on ColorPicker and ColorView, despite ColorPicker inherit from ColorView in the UI each one is defining their own view... make sense if I apply the style to the ColorView this is not affecting to the ColorPicker.

I ran into the same type of issue developing the ColorPicker... and I couldn't solve them. There are only work-arounds used for now and some functionality disabled.

In fact, if this issue didn't exist I think the entire ColorPicker control template would not have to have been duplicated.

I'm going to test a few things again with the Binding system refactor. Otherwise, I think this is just a known "bug" or area where Avalonia has missing features. I still don't understand it completely myself.

**Comment by ClayYahuira** [comment]
> > > Can you please also try on ColorView? I think it might relate to popup.
> > 
> > 
> > I tried aplying the style to the ColorView but it just work for the ColorView, this is not aplying to the ColorView in the popup.
> > I was checking the .xaml file on ColorPicker and ColorView, despite ColorPicker inherit from ColorView in the UI each one is defining their own view... make sense if I apply the style to the ColorView this is not affecting to the ColorPicker.
> 
> I ran into the same type of issue developing the ColorPicker... and I couldn't solve them. There are only work-arounds used for now and some functionality disabled.
> 
> In fact, if this issue didn't exist I think the entire ColorPicker control template would not have to have been duplicated.
> 
> I'm going to test a few things again with the Binding system refactor. Otherwise, I think this is just a known "bug" or area where Avalonia has missing features. I still don't understand it completely myself.

Do you know any workaround for this issue?

**Comment by robloo** [comment]
Worse-case you can re-template. There is probably a simpler work-around but I'm not sure at the moment.

Basically, the Popup content is materialized only when opened. And the styling system doesn't re-evaluate at that point. So since the Pop-up content doesn't exist when styles are applied... styles aren't set in popup content like the ColorPicker.

To work around that you have to set styles when the pop-up is opened, not before. That could be done in code-behind with a few tricks as well.

**Comment by timunie** [comment]
I honestly think it's better to try to fix the styling system instead of searching a workaround

**Comment by robloo** [comment]
@timunie I agree with you. However, this is well outside my area of knowledge. When it came up during the ColorPicker development you also noted that these types of life cycle issues are also present in WPF (I didn't remember them in UWP). At the time no one had a really good idea where to look or what exactly was going on in Avalonia.

**Comment by timunie** [comment]
It wasn't a message to you @robloo 

possible workaround is an InvalidateVisual call when Popup opens though.

**Comment by timunie** [comment]
was trying the binding refractor PR #13970 it doesn't solve the issue (yet) but it will throw an exception second time one opens the CP. 

```
Unhandled exception. System.InvalidOperationException: BindingExpression has not been started.
   at Avalonia.Data.Core.UntypedBindingExpressionBase.GetValue() 
```
